src/Chapter3.hs

```diff
@@ -479,6 +551,52 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+data Castle = NoCastle | CityCastle String deriving (Show)
+
+data Walls = NoWalls | CityWalls deriving (Show)
+
+data Hall = Church | Library deriving (Show)
+
+data House = BuildHouse 
+    {
+      housePoeple :: Int
+    } deriving (Show)
+
+data City = BuildCity
+    { cityCastle :: Castle
+    , cityHall :: Hall
+    , cityWalls :: Walls
+    , cityHouses :: [House]
+    } deriving (Show)
+
+type CastleName = String
+
+buildCastle :: City -> CastleName -> City
+buildCastle city name = city { cityCastle = (CityCastle name) }
+
+buildHouse :: City -> Int -> City
+buildHouse city numberOfPeople = city {cityHouses = freshHouse : (cityHouses city)}
+  where
+    freshHouse = BuildHouse numberOfPeople
```

Comment:
In this function, it is possible to pass any `Int`, but according to the task. houses should allow only from 0 to 4 people can live in a single house